### PR TITLE
fix: Ticks overlap 

### LINKED ISSUE #115: Ticks optimizations
We should prevent any overlap of ticks even with small charts while increase the number of ticks on large charts

### PR DESCRIPTION
**Pull request by NadjibSb**
closes #115

### CODE REVIEW

File: src/components/Axis/BottomAxis.tsx [hunk 1]
```diff
@@ -22,10 +22,22 @@ export default function BottomAxis({
...
+    const fontSizeDefault = 16;
+    const scientificTickLength = 7;
+    let tickLength = `${Math.trunc(scale?.domain()[1])}`.length;
+    // if domaine too small => tickLength+2 for decimal values
+    tickLength =
+      Math.abs(scale?.domain()[1] - scale?.domain()[0]) < plotWidth * 0.05
+        ? tickLength + 2
+        : tickLength;
+
+    const ticksNbr = scientific
```
Comment (maasencioh) [inline src/components/Axis/BottomAxis.tsx:35]:
```suggestion
    const ticksNumber = scientific
```

**Comment by targos** [comment]
There's a lot of similarity between the two changed pieces of code. Could you factor that out a bit ?

File: src/components/Axis/TopAxis.tsx [hunk 2]
```diff
@@ -22,10 +22,22 @@ export default function TopAxis({
...
+    const fontSizeDefault = 16;
+    const scientificTickLength = 7;
+    let tickLength = `${Math.trunc(scale?.domain()[1])}`.length;
+    // if domaine too small => tickLength+2 for decimal values
+    tickLength =
+      Math.abs(scale?.domain()[1] - scale?.domain()[0]) < plotWidth * 0.05
+        ? tickLength + 2
+        : tickLength;
+
+    const ticksNbr = scientific
```
Comment (maasencioh) [inline src/components/Axis/TopAxis.tsx:35]:
```suggestion
    const ticksNumber = scientific
```

**Comment by NadjibSb** [comment]
> There's a lot of similarity between the two changed pieces of code. Could you factor that out a bit ?

You mean between BottomAxis & TopAxis ? it's the same yeah
Should I extract the code that calculate Ticks number as util func ? or what do u suggest

**Comment by targos** [comment]
> Should I extract the code that calculate Ticks number as util func ?

Yes, please.

**Review by targos — APPROVED** [review]

**Pull request merged**